packages/pigeon/lib/ast.dart

```diff
@@ -105,6 +108,24 @@ class TypeDeclaration {
   /// True if the type is nullable.
   final bool isNullable;
 
+  @override
+  int get hashCode {
+    // This has to be implemented because TypeDeclaration is used as a Key to a
+    // Map in generator_tools.dart.
+    int hash = 17;
+    hash = hash * 37 + baseName.hashCode;
+    hash = hash * 37 + isNullable.hashCode;
+    for (final TypeDeclaration typeArgument in typeArguments) {
+      hash = hash * 37 + typeArgument.hashCode;
+    }
+    return hash;
+  }
+
+  @override
+  bool operator ==(Object other) {
+    return hashCode == other.hashCode;
```

Comment:
See https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#common-boilerplates-for-operator--and-hashcode

```suggestion
  if (other.runtimeType != runtimeType)
    return false;
  return other is TypeDeclaration
      && other.baseName == baseName
      && other.isNullable == isNullable
      && listEquals(other.typeArguments, typeArguments);
```